test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -1862,6 +1877,50 @@ def test_cgroup_cpuset_mem_hardwall(self):
         self.assertEqual(result['rc'], 0)
         self.assertEqual(result['out'][0], '1')
 
+    def test_cgroup_find_gpus(self):
+        """
+        Confirm that the hook finds the correct number
+        of GPUs.
+        """
+        if not self.paths['devices']:
+            self.skipTest('Skipping test since no devices subsystem defined')
+        name = 'CGROUP3'
+        self.load_config(self.cfg2)
+        cmd = ['nvidia-smi', '-L']
+        try:
+            rv = self.du.run_cmd(cmd=cmd)
+        except OSError:
+            rv = {'err': True}
+        if rv['err'] or 'GPU' not in rv['out'][0]:
+            self.skipTest('Skipping test since nvidia-smi not found')
+        gpus = int(len(rv['out']))
+        if gpus < 1:
+            self.skipTest('Skipping test since no gpus found')
+        self.server.expect(NODE, {'state': 'free'},
+                           self.hostA.shortname)
+        ngpus = self.server.status(NODE, 'resources_available.ngpus',
+                                   id=self.hostA.shortname)[0]
+        ngpus = int(ngpus['resources_available.ngpus'])
+        self.assertEqual(gpus, ngpus)
+        a = {'Resource_List.select': '1:ngpus=1', ATTR_N: name}
+        j = Job(TEST_USER, attrs=a)
+        j.create_script(self.check_gpu_script)
+        jid = self.server.submit(j)
+        a = {'job_state': 'R'}
+        self.server.expect(JOB, a, jid)
```

Comment:
You don't need to use 'a' if you don't want to.  It will all fit under 79 characters.  Up to you.